extmod/modlwip.c

```diff
@@ -720,6 +720,8 @@ STATIC mp_obj_t lwip_socket_listen(mp_obj_t self_in, mp_obj_t backlog_in) {
     socket->pcb.tcp = new_pcb;
     tcp_accept(new_pcb, _lwip_tcp_accept);
 
+    socket->state = STATE_CONNECTING;
```

Comment:
Arguable, a listening socket is still "new" and not yet connecting because there's not necessarily anyone trying to connect to it.  What problem does this line solve?  Is there a test case that can be written which shows why this line should be here?

Reply:
A listening socket will never be in other state other than new, but it should not return HUP or ERR anyway. Another condition will set HUP if the connection is not CONNECTING nor CONNECTED either.

Please find my simple test script here: https://gist.github.com/rkojedzinszky/cbb6710c0cd6b205eda6f32b4bfcd7fc